src/detect.c

```diff
@@ -865,8 +865,9 @@ static inline void DetectRulePacketRules(
                     SCLogDebug("no match in stream, fall back to packet payload");
 
                     /* skip if we don't have to inspect the packet and segment was
-                     * added to stream */
-                    if (!(sflags & SIG_FLAG_REQUIRE_PACKET) && (p->flags & PKT_STREAM_ADD)) {
+                     * added to stream or if the signature requires a stream. */
+                    if ((!(sflags & SIG_FLAG_REQUIRE_PACKET) && (p->flags & PKT_STREAM_ADD))
+                         || (sflags & SIG_FLAG_REQUIRE_STREAM)) {
```

Comment:
The line 849 checks (sflags & SIG_FLAG_REQUIRE_STREAM) != 0, thus this test will *always* branch to next.

Reply:
Thanx ... gdb showed me the a possible test  by looking to packet flags

p->flags = 4260096

```
#define PKT_HAS_FLOW                    (1<<8)

#define PKT_ZERO_COPY                   (1<<16)     /**< Packet comes from zero copy (ext_pkt must not be freed) */
#define PKT_WANTS_FLOW                  (1<<22)


```